src/order/create-order.ts

```diff
@@ -56,24 +57,9 @@ export default async function createOrder(
       body: JSON.stringify(orderPayload),
     });
 
-    const data = await response.json();
+    const data = (await response.json()) as CreateOrderResponse;
 
-    if (response.status !== 200 && response.status !== 201) {
-      const errorData = data as CreateOrderErrorResponse;
-
-      if (!errorData.name) {
-        throw new Error(defaultErrorMessage);
-      }
-
-      const { name, message, debug_id, details } = errorData;
-      const errorMessage = `${name} - ${message} (debug_id: ${debug_id})`;
-
-      const error: HttpErrorResponse = new Error(errorMessage);
-      error.details = details;
-      throw error;
-    }
-
-    return data as CreateOrderResponse;
+    return { data, httpStatus: response.status };
```

Comment:
I think the type could be `HttpErrorResponse` at this point. The `fetch()` helper does not assume that a 400 status code would be an error and go to the `catch()`. Previously we had to explicitly check the status code to know if the response shape is `CreateOrderResponse` or `HttpErrorResponse`. I would like these functions to continue returning the correct type of the payload. Perhaps we add back the statusCode checks and do the casting from there.

Ex:
```ts
const data = await response.json();

if (response.status !== 200 && response.status !== 201) {
  return { data: data as CreateOrderResponse, httpStatus: response.status }
else {
  return { data: data as CreateOrderErrorResponse, httpStatus: response.status }
}

Reply:
At this level (`/orders/` API files), my feeling is data should be pass-through without checking the status or shape. It will always be `CreateOrderResponse` as set on line 60 unless there is a network error. The reason for this is the controller has to turn around and (will by default) regurgitate the entire response with same status anyway, so we have it handle non-network errors.  I'm certainly open to being persuaded about why a CreateOrderErrorResponse & other error types would be helpful, but I'm not seeing it, to me they just confuse what's going on